### PR TITLE
[BugFix] Precast MoE gate weight_fp32 to avoid aclop Cast

### PR DESCRIPTION
**Pull request by LQDLove**
### What this PR does / why we need it?

Nightly `qwen3-vl-235b-a22b-instruct-w8a8` (A3, `FULL_DECODE_ONLY` ACLGraph) fails because internal-router MoE recomputes logits with:

`gate.weight.to(torch.float32)`

That emits a per-forward `aclop Cast`. DeepSeek-V4 and 310P already avoid this by setting `gate.precast_fp32_weight = True` so `AscendUnquantizedLinearMethod.process_weights_after_loading` materializes `weight_fp32` at load time.

This PR applies the same precast in `AscendMoERunner.__init__` for all Ascend platforms, and routes the hot path through `_gate_weight_fp32()` so ACLGraph capture no longer depends on a live Cast.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

- Code-path review against 310P `AscendMoERunner310` and DeepSeek-V4 `precast_fp32_weight`.
- Nightly should re-run `Qwen3-VL-235B-A22B-Instruct-W8A8`.

Made with [Cursor](https://cursor.com)

- vLLM main: https://github.com/vllm-project/vllm/commit/a97dacb7106ee49f39f3d1fc6ae1800ff724e01d

### CODE REVIEW

**Comment by czydyy** [comment]
/nightly qwen3-vl-235b-a22b-instruct-w8a8 
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/34431440206)

File: vllm_ascend/ops/fused_moe/fused_moe.py [hunk 1]
```diff
@@ -146,6 +154,18 @@ def is_internal_router(self) -> bool:
         # The runner must recompute router_logits via the gate.
         return self.gate is not None
 
+    def _gate_weight_fp32(self) -> torch.Tensor:
+        """Return the internal-router gate weight in FP32 without a hot-path Cast."""
+        gate = self.gate
+        assert gate is not None
+        weight_fp32 = getattr(gate, "weight_fp32", None)
```
Comment (lijiahang226) [inline vllm_ascend/ops/fused_moe/fused_moe.py:161]: Should we avoid the forward-time fallback cast here? If weight_fp32 is unexpectedly missing during ACLGraph capture, this can still reproduce the Cast issue this PR is trying to fix. Would it be safer to require weight_fp32 after weight processing and fail fast otherwise?

**Comment by czydyy** [comment]
/nightly Qwen3-32B-QuaRot multi-node-qwen3-dp multi-node-dpsk3.2-2node
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/34453663991)

**Comment by czydyy** [comment]
/nightly  multi-node-qwen3-dp multi-node-dpsk3.2-2node
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/34468870463)

File: vllm_ascend/ops/fused_moe/fused_moe.py (hunk 1 above)
Comment (LQDLove) [reply vllm_ascend/ops/fused_moe/fused_moe.py]: ok

**Review by weijinqian0 — DISMISSED** [review]

File: vllm_ascend/ops/fused_moe/fused_moe.py [hunk 2]
```diff
@@ -146,6 +154,18 @@ def is_internal_router(self) -> bool:
         # The runner must recompute router_logits via the gate.
         return self.gate is not None
 
+    def _gate_weight_fp32(self) -> torch.Tensor:
+        """Return the internal-router gate weight in FP32 without a hot-path Cast."""
```
Comment (weijinqian0) [inline vllm_ascend/ops/fused_moe/fused_moe.py:158]: Plz move this to __init__.

Comment (LQDLove) [reply vllm_ascend/ops/fused_moe/fused_moe.py]: 存在风险，在 init 里直接 gate.weight.to(fp32)：那时 Parameter 已在、权重未 load，会冻住随机初值。该方法已经再设备上验证无问题。功能正常和性能有恢复

**Comment by czydyy** [comment]
/nightly qwen3-vl-235b-a22b-instruct-w8a8
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/34679475397)

File: vllm_ascend/ops/fused_moe/fused_moe.py [hunk 3]
```diff
@@ -116,6 +116,12 @@ def __init__(
             self.moe_config.ep_group = get_ep_group()
             self.moe_config.mc2_group = get_mc2_group()
 
+        # Internal-router: seed weight_fp32 + precast for load refresh (avoid hot-path Cast).
+        # Use ctor `gate` (not self.is_internal_router): Module.__getattr__ shadows during init.
+        if gate is not None and not hasattr(gate, "weight_fp32"):
+            gate.precast_fp32_weight = True
```
Comment (weijinqian0) [inline vllm_ascend/ops/fused_moe/fused_moe.py]: plz remove this variable.

**Comment by czydyy** [comment]
/nightly qwen3-vl-235b-a22b-instruct-w8a8
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/34688314511)

**Comment by LQDLove** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by LQDLove** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Comment by czydyy** [comment]
/nightly qwen3-vl-235b-a22b-instruct-w8a8  multi-node-qwen3-dp
nightly command triggered.
- Nightly-A3: [workflow run](https://github.com/vllm-project/vllm-ascend/actions/runs/34734114394)

**Review by weijinqian0 — APPROVED** [review]
